networkx/algorithms/coloring/tests/test_sampled_recursive_largest_first.py

```diff
@@ -0,0 +1,108 @@
+"""sampled recursive largest first test suite.
+
+"""
+import pytest
+
+import networkx as nx
+
+is_coloring = nx.algorithms.coloring.equitable_coloring.is_coloring
+
+
+# ############################  Graph Generation ############################
+
+
+def empty_graph():
+    return nx.Graph()
+
+
+def one_node_graph():
+    graph = nx.Graph()
+    graph.add_nodes_from([1])
+    return graph
+
+
+def two_node_graph():
+    graph = nx.Graph()
+    graph.add_nodes_from([1, 2])
+    graph.add_edges_from([(1, 2)])
+    return graph
+
+
+def three_node_clique():
+    graph = nx.Graph()
+    graph.add_nodes_from([1, 2, 3])
+    graph.add_edges_from([(1, 2), (1, 3), (2, 3)])
+    return graph
+
+
+def disconnected():
+    graph = nx.Graph()
+    graph.add_edges_from([(1, 2), (2, 3), (4, 5), (5, 6)])
+    return graph
+
+
+def directed():
+    graph = nx.DiGraph()
+    graph.add_edges_from([(1, 2), (1, 3), (3, 4), (2, 5)])
+    return graph
+
+
+# --------------------------------------------------------------------------
+# Basic tests for all strategies
+# For each basic graph function, specify the number of expected colors.
+BASIC_TEST_CASES = {
+    empty_graph: 0,
+    one_node_graph: 1,
+    two_node_graph: 2,
+    disconnected: 2,
+    three_node_clique: 3,
+}
+
```

Comment:
I think the BASIC_TEST_CASES could be specified more clearly and compactly by creating objects rather than functions which create the objects. The test code doesn't modify the graphs so even if we add tests which reuse this list of basic tests, we should be OK using objects.

I'll also suggest that you consider changing the name of `n_nodes` to `n_colors` because the value for e.g. `disconnected` is definitely not the number of nodes.

Here's how I'm suggesting we implement objects rather than functions:
```python
BASIC_TEST_CASES = [
    nx.Graph(),  # empty graph
    nx.path_graph([1]),  # one_node_graph
    nx.Graph([(1, 2)]),  # single_edge_graph
    nx.cycle_graph(3),   # triangle (three_node_clique)
    nx.Graph([(1, 2), (2, 3), (4, 5), (5, 6)]),  # disconnected graph
]
```
Then in the test for directed graphs, hardcode in the directed graph `graph = nx.DiGraph([(1, 2), (2, 3), (4, 5), (5, 6)])`

I'm not saying it has to be this way, but its my suggestion.